sig-apps/Charter.md

```diff
@@ -0,0 +1,141 @@
+# SIG Apps Charter
+
+## Scope
+
+SIG Apps Covers deploying and operating applications in Kubernetes with a focus on the application developer and application operator experience.
+
+This includes but is not limited to:
+
+* Discussions on all levels from application development, testing environments, and production
+* Discussion of how to define and run apps in Kubernetes
+* Demos of relevant tools and projects
+* Discussion on areas of friction that can lead to suggesting improvements or feature requests
+* Development of Kubernetes primitives (e.g., Workloads API) to enable the application developers and operators. Note, most new types should be created as CRDs during their development phase and may always be ecosystem projects
+* Creation of documentation to enable application developers and operators
+* Development and maintenance of the sub-projects of Helm, Charts, and Kompose. Note, these projects are grandfathered and treated as part of the ecosystem but owned by SIG Apps. New projects are expected to stay in the ecosystem.
```

Comment:
What does the Kubernetes project gain by continuing to govern these projects? Arguably these projects did not belong in Kubernetes in the first place, and it seems that the authors agree by not accepting any new tools. This charter might be a good opportunity to make this split as Helm and charts seems to be self-sufficient to continue on their own.

* Does Helm v3 get "grandfathered in" if it is a complete rewrite of Helm v2?
* This is not the full list of sub-projects, does this mean SIG Apps is divesting the ones not listed here?

chart-testing
chartmuseum
charts (listed)
helm (listed)
kompose (listed)
monocular
rudder-federation

By being under SIG Apps governance, these tools are official tool recommendations of Kubernetes which runs counter to the non-goal of "endorsing a particular ecosystem tool"

Reply:
I agree with @r2d4  that we should have a clean separation here in terms of projects exactly because of the non-endorsing principle and that now, with this charter, it is the most appropriate point in time to implement it.

Reply:
This is a complicated space to discuss and conversations around this have been happening for months. Here are some of the details from those other conversations:

* Any change to the ownership of these subprojects is a steering committee decision. They have been talking about it and will continue to talk about it. The steering committee has a list of priorities and different elements relating to this have come up and will come up.
* Per the steering committee, all code must be owned by a SIG and have that recorded. That means that as long as Kubernetes owns these a SIG has to be responsible for them. SIG Apps is the current owner. This chartering process isn't attempting to change that.
* The Kubernetes project, which is part of the CNCF, which is part of the Linux Foundation owns these codebases. They are responsible for them and the health of them. They have a responsibility to figure out the healthy and appropriate path forward for them.
* Kubernetes is becoming more focused on the core. The things around the core could all be open to general change policies and there's debate on that. For example, in a general change policy what happens to minikube and dashboard which are not part of the core? A general change that affects Helm and Kompose would also look like it could apply to them as well. So, care is being taken in how any of this is handled.

From a practical standpoint, even though SIG Apps owns these we rarely talk about them in the primary meeting. We attempt to treat them like other ecosystem projects. The development of these projects isn't discussed in the main meeting or mailing list in a way that's different from the development of other ecosystem projects. They have their own meetings and lists to discuss their development.

I expect changes to come. I think there are better ways than what we do today. I'm actively working on some of that. When changes are ready, and have been worked out in the appropriate places, I expect we and others will be amending charters.

Reply:
Since these projects are rarely talked about in SIG Apps, then it would have little impact to remove them from the SIG Apps charter.

They can remain CNCF projects without being governed by SIG Apps, so legal ownership is not affected.

Reply:
@r2d4 per the steering committee, all subprojects (and all code) must be owned by a SIG. We could remove them from the charter but still have SIG Apps have ownership. The only way for code to still be owned by k8s but be removed from a SIGs ownership is for another SIG to pick it up or for it to be retired altogether. These projects are all too active to be retired both in terms of end users and in terms of contributors. The owned by k8s issue is an exercise separate from the process to add a charter because of who owns those kinds of decisions and the elements involved in it.

For another example, minikube cannot be dropped by SIG cluster lifecycle without another SIG picking it up or work done by the steering committee with the maintainers of minikube to land the project ownership somewhere other than k8s. That's a different level task than SIG Cluster Lifecycle creating their charter.

SIG Apps has been used as an example of a SIG that can manage a bit of organizational complexity. It has been able to own some subprojects while enabling them to mostly run independently of the weekly meetings. In practice it's worked and we'd like to see each of these projects grown to be able to function well as an open source project.

Reply:
I'm proposing that they are removed as subprojects as part of this charter. Since this charter claims this SIG has the power to add subprojects, it follows that it has the power to remove them.

They would still be owned by the CNCF, which does not require them to be owned by a Kubernetes SIG or "retired".

Reply:
That is a good question, but afaik there is no effort underway to move existing projects out of kubernetes. We should not disrupt current projects because we find the timing to be expedient. You could raise a larger issue with SIG governance if you feel there is value in taking inventory in what is currently part of currently governed and reducing that scope.

Reply:
A subproject of Kubernetes **_MUST_** be owned by a SIG. For something to shift out of Kubernetes, not needing to be owned by a SIG, but still be in the CNCF requires other governance processes occur. That's not the scope of this current PR and ask of the SIG which is to have a charter.

I am in favor of subprojects such as minikube, helm, and others that are not part of kubernetes core deliverable shifting around organizationally and in the CNCFs overall governance.

Because the process scope for this PR is for SIG Apps to get a charter and making changes to how subprojects associate with Kubernetes and the CNCF is both a different scope and one with other processes along with it requiring a whole lot more conversations and time I would suggest we don't block on that.

If you're not familiar, the relationship of subprojects not part of core Kubernetes and how they fit in the governance both now and long term has had hours of discussions by numerous people including the steering committee. If this blocks on that issue it will block SIG Apps, and possibly other SIGs, from getting a charter for some time.

Open source governance, especially at the scale of Kubernetes with people and vendors, is complicated and can take time to get setup well. The steering committee has a lot to do and part of that is even navigating priorities of what to do. While this isn't a low priority it doesn't appear to be their highest priority right now. That's ok and we can be a little patient. There are also pragmatic things, outside the current governance asks we have right now, that can be worked on and in some cases already are.

If anyone wants to discuss this in further detail I'm happy to.

Reply:
Subproject creation and removal is indeed part of this charter

https://github.com/mattfarina/community/blob/a5876519d816b84ca2dc65d0c49c6d0f17cdd74b/sig-apps/Charter.md#project-management

If this isn't actually a SIG Apps decision, then that governance should be moved into a higher level document and not this document.

Otherwise it follows that sub-project addition and removal is within the power of the governing SIG.

> I am in favor of subprojects such as minikube, helm, and others that are not part of kubernetes core deliverable shifting around organizationally and in the CNCFs overall governance.

 It seems like the co-chairs of SIG Apps also favor the removal of these projects, and it is well within their governing power to do so. I don't see any blockers for this to proceed.

Reply:
Thats fair, thanks for the clarification. I'll consider my comments resolved.